src/irr/asset/CGLILoader.cpp

```diff
@@ -48,284 +48,385 @@ namespace irr
-						// imagesOnAMipMapLevelToHandle[layer][face]->size = texture.size(layer);
-						// rawImagesBundle = imagesOnAMipMapLevelToHandle[0][0].get();    it is wrong, but we need a way to bind rawImagesBundle and imagesOnAMipMapLevelToHandle in a way iterating through it will be valid
+					{
+						auto index = (layer * texture.layers()) + face;
+						texelBuffers.emplace_back(core::make_smart_refctd_ptr<ICPUBuffer>(baseImage->getImageDataSizeInBytes())); // I need to put data here, but
+						// core::smart_refctd_ptr<CCustomAllocatorCPUBuffer<>> buffer = CCustomAllocatorCPUBuffer<>(XSIZEX, XDATAX);	is inaccessable!
+
+						ICPUImage::SBufferCopy& region = *(regions->begin() + index);
+						region.imageSubresource.mipLevel = 0u;
+						region.imageSubresource.baseArrayLayer = 0u;
+						region.imageSubresource.layerCount = 1u;
+						region.bufferOffset = 0u;
+						region.bufferRowLength = extent[0] * texelFormatByteSize;
+						region.bufferImageHeight = 0u; //tightly packed
+						region.imageOffset = { 0u, 0u, 0u };
+						region.imageExtent = {extent[0], extent[1], extent[2]};
 					}
+
+				auto& viewImage = core::make_smart_refctd_ptr<ICPUImageView>(*(bundle.second->begin() + mipmapLevel));
+				// TODO need to set regions on viewImage as subresource of certain mipmap
+				viewImage = ICPUImageView::create(std::move(imageViewInfo));
 			}
 
 			ctx.file = nullptr;
 
-			return SAssetBundle({ core::smart_refctd_ptr<IAsset>(rawImagesBundle, core::dont_grab) });
+			//return SAssetBundle( ? );
 		}
 
 		bool CGLILoader::isALoadableFileFormat(io::IReadFile* _file) const
 		{
 			return true; // gli provides a function to load files, but we can check files' signature actually if needed
 		}
 
-		inline E_FORMAT CGLILoader::getTranslatedGLIFormat(const gli::texture& texture, const gli::gl& glVersion)
+		inline std::pair<E_FORMAT, ICPUImageView::SComponentMapping> CGLILoader::getTranslatedGLIFormat(const gli::texture& texture, const gli::gl& glVersion)
 		{
 			using namespace gli;
 			gli::gl::format formatToTranslate = glVersion.translate(texture.format(), texture.swizzles());
+			ICPUImageView::SComponentMapping compomentMapping;
+
+			constexpr std::array<std::pair<gli::gl::swizzle, ICPUImageView::SComponentMapping::E_SWIZZLE>, 6> swizzlesMappingAPI =
+			{
+				std::make_pair(gl::SWIZZLE_RED, ICPUImageView::SComponentMapping::ES_R),
+				std::make_pair(gl::SWIZZLE_GREEN, ICPUImageView::SComponentMapping::ES_G),
+				std::make_pair(gl::SWIZZLE_BLUE, ICPUImageView::SComponentMapping::ES_B),
+				std::make_pair(gl::SWIZZLE_ALPHA, ICPUImageView::SComponentMapping::ES_A),
+				std::make_pair(gl::SWIZZLE_ONE, ICPUImageView::SComponentMapping::ES_ONE),
+				std::make_pair(gl::SWIZZLE_ZERO, ICPUImageView::SComponentMapping::ES_ZERO)
+			};
+
+			auto getMappedSwizzle = [&](const gli::gl::swizzle& currentSwizzleToCheck)
+			{
+				for (auto& mappedSwizzle : swizzlesMappingAPI)
```

Comment:
this is stupid, `gl::SWIZZLE` is an enum, you can just make an array and just index it by the swizzle

Reply:
sure, but it seems to me it won't match because of that:
https://github.com/g-truc/gli/blob/master/gli/gl.hpp#L314

Reply:
use `core::unordered_map` then

Reply:
a `static const core::unordered_map`